homeassistant/components/bloomsky/sensor.py

```diff
@@ -60,7 +60,7 @@
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the available BloomSky weather sensors."""
     # Default needed in case of discovery
-    sensors = config[CONF_MONITORED_CONDITIONS]
+    sensors = config.get(CONF_MONITORED_CONDITIONS, SENSOR_TYPES)
```

Comment:
I don't understand, don't we have a platform schema?

Reply:
Yes we do... which is the weird part

Reply:
Oh I see what is happening. Bloomsky component loads this with a discovery and an empty dictionary. Instead of doing this, we should guard it to not run when discovery info is not None for all 3 platforms.
```suggestion
    if discovery_info is not None:
        return

    sensors = config[CONF_MONITORED_CONDITIONS]
```

Reply:
When we set up a platform, we always first make sure we set up the component, which would then discover this.

Reply:
Why do we load the platforms via discovery at all if we don't allow the platform setup to complete in that case?